src/lib/AppState.tsx

```diff
@@ -4,20 +4,29 @@ import { useContext } from 'preact/hooks';
 import { RecyclingLocatorAttributes } from '..';
 
 interface AppStateContext extends RecyclingLocatorAttributes {
-  startRoute: string;
+  startPath: string;
 }
 
 export const AppState = createContext<AppStateContext>(null);
 
+function getStartPath(attributes: RecyclingLocatorAttributes): string {
+  const { postcode } = attributes;
+
+  if (attributes.path) {
+    return attributes.path;
+  } else if (attributes.postcode) {
```

Comment:
We don't need the `else if` because we have a return in the previous `if`, so we can simplify it to just `if`.

It's probably also worth moving this out to lib and writing unit tests.